packages/components/src/slot-fill/context.js

```diff
@@ -109,19 +108,13 @@ class SlotFillProvider extends Component {
 		if ( this.slots[ name ] !== slotInstance ) {
 			return [];
 		}
-		return sortBy( this.fills[ name ], 'occurrence' );
```

Comment:
What problem were we solving earlier that we needed to count instances?

Reply:
It's very difficult to tell even when you go through past commits and related discussions. There was definitely something that is covered with the unit test I updated. It was discussed in some of the PRs I studied but there wasn't a clear answer why this particular test needs to be kept:
https://github.com/WordPress/gutenberg/blob/9111ad51e3f123b0b60cc75666412991e834a46b/packages/components/src/slot-fill/test/slot.js#L170-L209

I extended this test to see how it behaves in the `trunk` branch, and I realized that the test is too specific and doesn't cover more complex cases. The updated version of the test produces the following order that is far from perfect:

```
<div>
    third
    first (rerendered)
    second
    fourth (new)
</div>
```

The code of the test:
https://github.com/WordPress/gutenberg/blob/255928e066cf530d6ea6e62ee71814cf45eb780d/packages/components/src/slot-fill/test/slot.js#L231-L273

Reply:
Curious, if we can't remember, it does make sense to update the expectation here

Reply:
<blockquote>

```
<div>
    third
    first (rerendered)
    second
    fourth (new)
</div>
```

</blockquote>

@gziolo Does this happen on the main branch or only on this PR? I think rendering `Fill`s conditionally (instead of just omitting its contents) like in that test is a pretty common thing.

Reply:
@diegohaz, I copied the result from the main branch. I wrongly assumed that it's easy to find the result in the diff section for the same test, that is in this branch:

```
<div>
     second
     third
     first (rerendered)
     fourth (new)
</div>
```

It's slightly better as more predictable, but it's still far from perfect. This is also why fills for `PinnedItems` are optimized to be always rendered, and they conditionally show its content. This way, the order doesn't change in my testing, and as I can tell, it can be confirmed by others.